stout/filesystem.h

```diff
@@ -15,6 +16,10 @@ namespace filesystem {
 
 ////////////////////////////////////////////////////////////////////////
 
+namespace detail {
+
+////////////////////////////////////////////////////////////////////////
+
 // Moveable, not Copyable.
 class Request {
```

Comment:
How about moving this into `EventLoop::Request` too?

Reply:
There are different types of requests in libuv. I can name at least two right now: uv_fs_t and uv_write_t. They are meant for different things and it's definitely not a good idea to move them into the EventLoop

Reply:
Ok, how about we create a `filesystem` namespace instead of `defailt` and move `Request` to `filesystem::Request` instead of `detail::Request`. `Request` is very generic and will pollute the `detail` namespace unless it's the only `Request` we ever imagine having in `stout::eventuals`, which is dubious. I also think we'll move more and more away from a generic `detail` namespace overtime. Thanks!

Reply:
It was previously `filesystem::Request` and this PR moves it into `filesystem::detail::Request`.